components/Registration/Registration.tsx

```diff
@@ -86,14 +90,170 @@ const RegistrationForm: React.FC = () => {
+            hackEssay1: "",
+            hackEssay2: "",
+            optionalEssay: "",
+            resumeFileName: "",
+            location: "",
+            gender: "",
+            degree: "",
+            major: "",
+            minor: "",
+            gradYear: 0,
+            isProApplicant: false,
+            proEssay: "",
+            considerForGeneral: false,
+            requestedTravelReimbursement: false,
+            dietaryRestrictions: [],
+            race: [],
+            hackInterest: [],
+            hackOutreach: []
+        };
+
+        if (registrationWithId) {
+            const { id, ...registration } = registrationWithId;
+            finalRegistrationObject = {
+                ...finalRegistrationObject,
+                ...registration
+            };
+        }
+
+        for (const key in finalRegistrationObject) {
+            if (key in fieldValues) {
+                if (key === "race") {
+                    const raceValue = fieldValues[key as keyof FieldValues];
+                    finalRegistrationObject.race = raceValue
+                        ? Array.isArray(raceValue)
+                            ? (raceValue as string[])
+                            : [raceValue as string]
+                        : [];
+                } else if (key === "gradYear") {
+                    const gradYearValue = fieldValues[key as keyof FieldValues];
+                    finalRegistrationObject.gradYear = Number(gradYearValue);
+                } else if (key === "requestedTravelReimbursement") {
+                    if (fieldValues[key as keyof FieldValues][0] === "YES") {
+                        finalRegistrationObject.requestedTravelReimbursement =
+                            true;
+                    }
+                } else {
+                    // console.log("key", key);
+                    finalRegistrationObject[key as keyof RegistrationType] =
+                        fieldValues[key as keyof FieldValues];
+                }
+            }
+        }
+
+        return finalRegistrationObject;
+    };
+
+    const onSubmit = async (
         values: FieldValues,
         formikHelpers: FormikHelpers<FieldValues>
     ) => {
-        console.log("submit", values);
+        // console.log("submit", values);
+
+        const registrationObject = await createRegistrationObject(values);
+
+        // console.log("full thing", registrationObject);
+
         handlePageChange(formIndex + 1);
+
+        registerUpdate(registrationObject)
+            // repsonse is not returning resumefileName even though registration object has the field, can't figure out why
+            .then(response => console.log("Response:", response))
+            .catch(error => {
+                console.error(
+                    "Error Response:",
+                    error.response?.data || error.message
+                );
+            });
     };
 
+    useEffect(() => {
+        getRegistration()
```

Comment:
Can this GET request be done only on first page load and then the registration object be kept up to date as the user fills in the form?